integrations-and-supported-tools/pytorch/scripts/Neptune_Pytorch_Support.py

```diff
@@ -0,0 +1,115 @@
+from numpy.random import permutation
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torchvision
+from torchvision import datasets, models, transforms
+import neptune.new as neptune
+import numpy as np
+
+
+def main():
+    # Step 1: Initialize Neptune and create new run
+    run = neptune.init(project='common/pytorch-integration', api_token='ANONYMOUS', source_files=['*.py'])
+
+    # Experiment Config
+    data_dir = 'data/CIFAR10'
+    compressed_ds = './data/CIFAR10/cifar-10-python.tar.gz'
+    data_tfms = {
+            'train': transforms.Compose([
+                transforms.RandomHorizontalFlip(),
+                transforms.ToTensor(),
+                transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+            ])
+        }
+
+    params = {
+        'lr': 1e-2,
+        'bs': 128,
+        'input_sz': 32 * 32 * 3,
+        'n_classes': 10,
+        'epochs': 10,
+        'model_filename': 'basemodel',
+    }
+
+    # Model & Dataset
+    class BaseModel(nn.Module):
+        def __init__(self, input_sz, hidden_dim, n_classes):
+            super(BaseModel, self).__init__()
+            self.main = nn.Sequential(
+                nn.Linear(input_sz, hidden_dim*2),
+                nn.ReLU(),
+                nn.Linear(hidden_dim*2, hidden_dim),
+                nn.ReLU(),
+                nn.Linear(hidden_dim, hidden_dim//2),
+                nn.ReLU(),
+                nn.Linear(hidden_dim//2, n_classes)
+            )
+
+        def forward(self, input):
+            x = input.view(-1, 32* 32 * 3)
+            return self.main(x)
+
+    trainset = datasets.CIFAR10(data_dir, transform=data_tfms['train'], download=True)
+    trainloader = torch.utils.data.DataLoader(trainset, batch_size=params['bs'], shuffle=True, num_workers=2)
+    dataset_size = {'train': len(trainset)}
+
+    # Instatiate model, crit & opt
+    model = BaseModel(params['input_sz'], params['input_sz'], params['n_classes'])
+    criterion = nn.CrossEntropyLoss()
+    optimizer = optim.SGD(model.parameters(), lr=params['lr'])
+
+    # Step 2: Log config & pararameters
```

Comment:
I think we should have those logging parts close to the places where they are created. 

For example params right below the params definition and so forth -> it will be easier to read/work with

Reply:
That is really hard to do with the way I envisioned this.

Which is in steps if you look at it the step is nicely laid in the code too.

Unless it is not important to create that structure I can do that.